tests/test_num_util.cc

```diff
@@ -146,6 +146,11 @@ TEST_CASE("test within_orders()", "[num_util]")
     REQUIRE(gul14::within_orders(1.0, 0.0, 1) == false);
     REQUIRE(gul14::within_orders(1.0, 0.0, 0) == false);
 
+    // Compare zero with zero
+    double a = 0.0;
+    REQUIRE(gul14::within_orders<double>(a, a, 5) == true);
+    REQUIRE(gul14::within_orders(static_cast<double>(0.0f), 0.0, 5) == true);
```

Comment:
The static_cast strikes me as weird: Why not directly test `within_orders(0.0, 0.0, 5) == true`?

Reply:
I wanted to have a cast from `float` to `double`, as that changes the internal representation and thus the value (sometimes).

Of course a zero should still be a zero ofterwards.

Maybe reflects better my intentions:

```suggestion
    float b = 0.0f;
    REQUIRE(gul14::within_orders(static_cast<double>(b), 0.0, 5) == true);
```

But then, maybe just rely on the fact that a `float` zero cast to `double` is the same as a direct `double` zero is.

This is not the case for some other numbers like 0.1, as they are represented with different rounding.
I.e. `0.1f != 0.1`.

Reply:
Changed code a bit.